middleware/index.js

```diff
@@ -1,39 +1,38 @@
-const Campground = require('../models/campground')
+const Article = require('../models/article')
 const Comment = require('../models/comment')
 //all the middleware goes here
 let middlewareObj = {}
 
-middlewareObj.checkCampgroundOwnership = (req, res, next) => {
-    if(req.isAuthenticated()){
-        Campground.findById(req.params.id, (err, foundCampground) => {
+middlewareObj.checkArticleOwnership = (req, res, next) => {
+    if(req.isAuthenticated()) {
+        Article.findById(req.params.id, (err, foundArticle) => {
             if(err) {
-                req.flash('error', 'Campground not found')
+                req.flash('error', 'Article not found :(')
                 return res.redirect('back')
             }
-            if(foundCampground.author.id.equals(req.user._id) || req.user.isAdmin)
-                return next()
 
-            req.flash('error', 'You don\'t have permission to do that')
-            return res.redirect('back')
-        })   
+            if(foundArticle.author.id.equals(req.user._id) || req.user.isAdmin)
+                return next()
+            
+            req.flash('error', 'You don\'t have permission to do that!')
+            return res.redirect
+        })
     }
-    req.flash("error", "You need to be signed in to do that!")
-    return res.redirect("/login")
 }
 
 middlewareObj.checkCommentOwnership = (req, res, next) => {
-    if(req.isAuthenticated()){
-        Comment.findById(req.params.comment_id, (err, comment) => {
-            if(comment.author.id.equals(req.user._id)|| req.user.isAdmin)
-                return next()
+    if(!req.isAuthenticated()){
+        req.flash('error', 'You need to be logged in to do that!')
+        return res.redirect('back')
+    }
 
+    Comment.findById(req.params.comment_id, (err, comment) => {
+        if(!comment.author.id.equals(req.user._id) || !req.user.isAdmin){
```

Comment:
I see that this check repeats, maybe we could include it as a middleware

Reply:
I feel like refactoring this would just make the codebase messier without providing any real improvement.